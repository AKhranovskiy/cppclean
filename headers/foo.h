
#include <map>
#include <string>

#include "bar.h"
#include "xss_module.h"


typedef std::map<std::string, int> Mapper;
typedef struct { int zz; } AnonStruct;
typedef union NamedUnion { bool xx; } *NamedPtr;

class Baz;
class Unused;
class Local;
class DataMember;
class Ptr;

class Data;
extern Data* d;

namespace somespace {
class Jj;
}

namespace FooNS {

class Foo : public Bar {
 public:
  explicit Foo();
  Foo(register const unsigned long& a, Baz* output) {}
  virtual ~Foo() = 0;

 protected:
  virtual const somespace::Jj& Get() const;
  const somespace::Jj& Get2() const = 0;
  int Get3() const { return 0; }

  template <class T>
  T& foo(T* bar);

 private:
  void Method(const Mapper& mapper, somespace::Jj* j);
  Mapper& Blah();
  Mapper data_;
  DataMember* data;
  auto_ptr<Ptr*> ptr;
};

void Function(const Foo& foo, Bar* bar);
extern const ws::RMM *full;
void Function2(const Local& local);

}

class FooModule : public XSS::Module {
 public:
  FooModule() : XSS::Module("mod") { }
};

template<class> class ProtoArray;

class Experiment {
  explicit Experiment();

  struct Foo {
    bool operator() (const Foo *a, const Foo *b) {
      return true;
    }
  };

 private:
};


namespace noname {
enum nn {
  bar,
};
};

namespace {
// Anon namespace
};

#include "bar.h"  // foo
#include <string>  // foo

class Abc::Xyz {};

class Fool : public Bar, public Baz {};


namespace {
// Anon namespace with something
void lkjsdfkd();
};

class TT : public TTbase<Bar> {};

void (*ptof)(const TT& tt);

typedef void(*Setter)(int*, int);

// filling data.
Fool data[] = { NULL, };

class def_var {
  static const int kConstant1 = 5;
  static const int kConstant = (kConstant2 - 1) / 4;
};

#include "not-used.h"

namespace {
class LaunchNameEquals {
 public: 
  bool operator()(void* launch) {
    return true;
  } 
};
}  // namespace

enum {boo};

class Blashdfdf {
 public: 
  bool operator()(void* launch) const = 0;
};

const char kBS[] = "~!@#$%^&*()-+{}|`[]\\:\";'<>?,./ =\t\r\n\v\f\177";

class FooOnYou {
  void (sq_type::*field_)(string&);
};

namespace wtf = noname;

void AnnotatedFunction() HOPEFULLY_THIS_IS_AN_ATTR;
void RealAnnotatedFunction() __attribute__((weakref));

class TestAttrs {
  void AnnotatedMethod() HOPEFULLY_THIS_IS_AN_ATTR;
  void RealAnnotatedMethod() __attribute__((weakref));
};

typedef struct _IplImage IplImage;

namespace {
struct AR;
class BS { };
}

class OpCode {
public:
   enum Operator {
    ID = 0,
    CASE = 1,
    DEL = 2,
    INS = 3,
   };
   enum Operator op;
};

class MemoryCache {
  mutable Mutex index_mutex_;
  struct CacheEntry **index_;
};

struct lala : dense_hashmap<int> {};

#if 0
#if 1
ignore more stuff a second apostrophe here does not work
#else
something else
#endif
#endif

void Atmak() throw(std::bull);

enum Status;
