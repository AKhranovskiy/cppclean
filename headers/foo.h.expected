Processing foo.h
Include(1, 15, <map>)
Include(16, 33, <string>)
Include(35, 51, "bar.h")
Include(52, 75, "xss_module.h")
Typedef(86, 89, Mapper, [Token('std', 86, 89), Token('::', 89, 91), Token('map', 91, 94), Token('<', 94, 95), Token('std', 95, 98), Token('::', 98, 100), Token('string', 100, 106), Token(',', 106, 107), Token('int', 108, 111), Token('>', 111, 112)])
Typedef(136, 137, AnonStruct, [Struct(136, 137, None, None, [VariableDeclaration(138, 141, int  zz)])])
Typedef(185, 186, NamedPtr, [Union(185, 186, NamedUnion, {[Token('bool', 187, 191), Token('xx', 192, 194), Token(';', 194, 195)]}), Token('*', 198, 199)])
Class(216, 219, Baz, None, None)
Class(227, 233, Unused, None, None)
Class(241, 246, Local, None, None)
Class(254, 264, DataMember, None, None)
Class(272, 275, Ptr, None, None)
Class(284, 288, Data, None, None)
VariableDeclaration(290, 296, extern Data* d)
Class(335, 337, Jj, None, None in somespace)
Class(367, 370, Foo, [[Token('Bar', 380, 383)]], [Function(406, 409, None Foo([]), 0x08, None), Function(415, 418, None Foo([Token('register', 419, 427), Token('const', 428, 433), Token('unsigned', 434, 442), Token('long', 443, 447), Token('&', 447, 448), Token('a', 449, 450), Token(',', 450, 451), Token('Baz', 452, 455), Token('*', 455, 456), Token('output', 457, 463)]), 0x00, []), Function(479, 482, None Foo([]), 0x16, None), Function(513, 518, [Token('const', 513, 518), Token('somespace', 519, 528), Token('::', 528, 530), Token('Jj', 530, 532), Token('&', 532, 533)] Get([]), 0x03, None), Function(549, 554, [Token('const', 549, 554), Token('somespace', 555, 564), Token('::', 564, 566), Token('Jj', 566, 568), Token('&', 568, 569)] Get2([]), 0x05, None), Function(590, 593, [Token('int', 590, 593)] Get3([]), 0x01, [Token('return', 609, 615), Token('0', 616, 617), Token(';', 617, 618)]), Function(645, 646, [Token('T', 645, 646), Token('&', 646, 647)] foo([Token('T', 652, 653), Token('*', 653, 654), Token('bar', 655, 658)]), 0x00, None), Function(674, 678, [Token('void', 674, 678)] Method([Token('const', 686, 691), Token('Mapper', 692, 698), Token('&', 698, 699), Token('mapper', 700, 706), Token(',', 706, 707), Token('somespace', 708, 717), Token('::', 717, 719), Token('Jj', 719, 721), Token('*', 721, 722), Token('j', 723, 724)]), 0x00, None), Function(729, 735, [Token('Mapper', 729, 735), Token('&', 735, 736)] Blah([]), 0x00, None), VariableDeclaration(747, 753,  Mapper data_), VariableDeclaration(763, 773,  DataMember* data), VariableDeclaration(783, 791,  Ptr* ptr)] in FooNS)
Function(807, 811, [Token('void', 807, 811)] Function([Token('const', 821, 826), Token('Foo', 827, 830), Token('&', 830, 831), Token('foo', 832, 835), Token(',', 835, 836), Token('Bar', 837, 840), Token('*', 840, 841), Token('bar', 842, 845)]), 0x00, None in FooNS)
VariableDeclaration(848, 854, extern const ws::RMM* full)
Function(876, 880, [Token('void', 876, 880)] Function2([Token('const', 891, 896), Token('Local', 897, 902), Token('&', 902, 903), Token('local', 904, 909)]), 0x00, None in FooNS)
Class(922, 931, FooModule, [[Token('XSS', 941, 944), Token('::', 944, 946), Token('Module', 946, 952)]], [Function(966, 975, None FooModule([]), 0x00, [])])
Class(1029, 1039, ProtoArray, None, None)
Class(1048, 1058, Experiment, None, [Function(1072, 1082, None Experiment([]), 0x08, None), Struct(1100, 1101, None, None, [Function(1096, 1099, [Token('Foo', 1096, 1099), Token('bool', 1106, 1110)] operator([Token('const', 1123, 1128), Token('Foo', 1129, 1132), Token('*', 1133, 1134), Token('a', 1134, 1135), Token(',', 1135, 1136), Token('const', 1137, 1142), Token('Foo', 1143, 1146), Token('*', 1147, 1148), Token('b', 1148, 1149)]), 0x00, [Token('return', 1159, 1165), Token('true', 1166, 1170), Token(';', 1170, 1171)])])])
Enum(1226, 1227, nn, {[Token('bar', 1230, 1233), Token(',', 1233, 1234)]} in noname)
Include(1276, 1294, "bar.h")
Include(1301, 1320, <string>)
Class(1334, 1337, Abc::Xyz, None, [])
Class(1354, 1358, Fool, [[Token('Bar', 1368, 1371)], [Token('Baz', 1380, 1383)]], [])
Function(1435, 1439, [Token('void', 1435, 1439)] lkjsdfkd([]), 0x00, None in <anonymous>)
Class(1462, 1464, TT, [[Token('TTbase', 1474, 1480), Token('<', 1480, 1481), Token('Bar', 1481, 1484)]], [])
VariableDeclaration(1491, 1495, * void ptof)
Typedef(1528, 1532, ), [None, Token('(', 1532, 1533), Token('*', 1533, 1534), Token('Setter', 1534, 1540), Token(')', 1540, 1541), Token('(', 1541, 1542), Token('int', 1542, 1545), Token('*', 1545, 1546), Token(',', 1546, 1547), Token('int', 1548, 1551)])
VariableDeclaration(1588, 1592,  NULL, })
Class(1604, 1611, def_var, None, [VariableDeclaration(1616, 1622, static const int kConstant1= 5), VariableDeclaration(1651, 1657, static const int kConstant=(kConstant2-1)/ 4)])
Include(1706, 1727, "not-used.h")
Class(1747, 1763, LaunchNameEquals, None, [Function(1778, 1782, [Token('bool', 1778, 1782)] operator([Token('void', 1794, 1798), Token('*', 1798, 1799), Token('launch', 1800, 1806)]), 0x00, [Token('return', 1814, 1820), Token('true', 1821, 1825), Token(';', 1825, 1826)])] in <anonymous>)
Enum(1857, 1858, None, {[Token('boo', 1858, 1861)]})
Class(1871, 1880, Blashdfdf, None, [Function(1895, 1899, [Token('bool', 1895, 1899)] operator([Token('void', 1911, 1915), Token('*', 1915, 1916), Token('launch', 1917, 1923)]), 0x05, None)])
VariableDeclaration(1940, 1945, const char kBS[]= "~!@#$%^&*()-+{}|`[]\\:\";'<>?,./ =\t\r\n\v\f\177")
Class(2018, 2026, FooOnYou, None, [VariableDeclaration(2031, 2035, string & void field_)])
Function(2094, 2098, [Token('void', 2094, 2098)] AnnotatedFunction([]), 0x40, None)
Function(2146, 2150, [Token('void', 2146, 2150)] RealAnnotatedFunction([]), 0x20, None)
Class(2208, 2217, TestAttrs, None, [Function(2222, 2226, [Token('void', 2222, 2226)] AnnotatedMethod([]), 0x40, None), Function(2274, 2278, [Token('void', 2274, 2278)] RealAnnotatedMethod([]), 0x20, None)])
Typedef(2346, 2355, IplImage, [Struct(2346, 2355, _IplImage, None, None)])
Struct(2386, 2388, AR, None, None in <anonymous>)
Class(2396, 2398, BS, None, [] in <anonymous>)
Class(2413, 2419, OpCode, None, [Enum(2447, 2448, Operator, {[Token('ID', 2453, 2455), Token('=', 2456, 2457), Token('0', 2458, 2459), Token(',', 2459, 2460), Token('CASE', 2465, 2469), Token('=', 2470, 2471), Token('1', 2472, 2473), Token(',', 2473, 2474), Token('DEL', 2479, 2482), Token('=', 2483, 2484), Token('2', 2485, 2486), Token(',', 2486, 2487), Token('INS', 2492, 2495), Token('=', 2496, 2497), Token('3', 2498, 2499), Token(',', 2499, 2500)]}), VariableDeclaration(2524, 2526,  Operator op)])
